PathFollow does not update when unit_offset set to 0

**Godot version:**


3.1.2.stable.custom_build

**OS/device including version:**

Linux 5.4.13-arch1-1

**Issue description:**

My game contains paths that trains follow. Every path has a tunnel at the start and the end.
I'd like to be able to modify the path and have the tunnels always stay at the start and the end.
I'm able to get the end tunnel to update, but not the start, as setting `PathFollow.unit_offset` to `0` does not seem to update the Transform.

**Steps to reproduce:**

1. Create the following script
```
tool
extends PathFollow
class_name PathPin

export(float, 0, 1) var pin := 0.0

onready var path: Path = get_parent()

func _ready():
	path.connect("curve_changed", self, "_on_curve_changed")

func _on_curve_changed():
	unit_offset = pin
```

2. Attach it to a pathfollow that is child to a Path
2a. Ensure `loop=false` otherwise it will [wrap back to the start](https://github.com/godotengine/godot/issues/7665) (probably fixed in 3.2)
3. Drag the first point of the path around
4. Note that the `PathFollow` does not update its transform to match the start of the path
5. Set `pin > 0`
6. Drag the path around
7. Note that the `PathFollow` now _does_ update its transform

**Minimal reproduction project:**
![1579872219](https://user-images.githubusercontent.com/2496231/73072333-df39d000-3e82-11ea-9c65-67beb851ad7f.png)
[example.zip](https://github.com/godotengine/godot/files/4108618/example.zip)

No luck building RC-3 or master.

```
main/default_controller_mappings.gen.cpp:3:10: fatal error: typedefs.h: No such file or directory
    3 | #include "typedefs.h"
```

Should that be `core/typedefs.h`?

--- a/scene/3d/path_3d.cpp
+++ b/scene/3d/path_3d.cpp
@@ -94,10 +94,6 @@ void PathFollow3D::_update_transform(bool p_update_xyz_rot) {
 		return;
 	}
 
-	if (delta_offset == 0) {
-		return;
-	}
-
 	float bl = c->get_baked_length();
 	if (bl == 0.0) {
 		return;
@@ -156,7 +152,7 @@ void PathFollow3D::_update_transform(bool p_update_xyz_rot) {
 
 		t.origin = pos;
 
-		if (p_update_xyz_rot) { // Only update rotation if some parameter has changed - i.e. not on addition to scene tree
+		if (p_update_xyz_rot && delta_offset != 0) { // Only update rotation if some parameter has changed - i.e. not on addition to scene tree.
 			Vector3 t_prev = (pos - c->interpolate_baked(offset - delta_offset, cubic)).normalized();
 			Vector3 t_cur = (c->interpolate_baked(offset + delta_offset, cubic) - pos).normalized();
 

